lib/Dist/Zilla/Plugin/UploadToCPAN.pm

```diff
@@ -221,6 +221,30 @@ has upload_uri => (
   predicate => 'has_upload_uri',
 );
 
+=attr retries
+
+The number of retries to perform on upload failure. This value will be passed to
+L<CPAN::Uploader>.
+
+=cut
+
+has retries => (
+  is => 'ro',
```

Comment:
These attributes should probably have a sensible default value, rather than being left to be undef. Something like 3 retries with a 10 s delay would be reasonable IMO.